### PR TITLE
Structural patterns

### PR DESCRIPTION
**Pull request by SiarheiBarkouski**
StructuralPatterns done.

### CODE REVIEW

File: StructuralPatterns/Adapter/StringAdapter.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+        public StringAdapter()
+        {
+            _inputString = String.Empty;
+        }
+
+        public string InputString
+        {
+            get => _inputString;
+            set => _inputString = value;
```
Comment (MaksimNikitin) [inline StructuralPatterns/Adapter/StringAdapter.cs]: You could try ti get rid of the filed and make the property auto. Intitializing form the cinstructor should work fine as well.

File: StructuralPatterns/Adapter/StringAdapter.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        }
+        public void CountLetters(params char[] letters)
+        {
+            int counter = 0;
```
Comment (MaksimNikitin) [inline StructuralPatterns/Adapter/StringAdapter.cs]: using of var is preferable

File: StructuralPatterns/Adapter/StringAdapter.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            Console.WriteLine("__________\n" + _inputString);
+            for (int i = 0; i < letters.Length; i++)
+            {
+                counter += _inputString.Count(t => t.Equals(letters[i]));
```
Comment (MaksimNikitin) [inline StructuralPatterns/Adapter/StringAdapter.cs]: Your method is supposed to return just a number. Do not need to display anything on console, it should be program.cs responcibility.

File: StructuralPatterns/Facade/DifficultMathOperations.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Facade
+{
+    class DifficultMathOperations
+    {
+        private static readonly Func<int, long> _factorial = x => x == 0 ? 1 : _factorial(x - 1) * x; 
+
+        public double Log(int a)
```
Comment (MaksimNikitin) [inline StructuralPatterns/Facade/DifficultMathOperations.cs]: The same thing as in the previous PR. Rework names of your methods.

**Review by MaksimNikitin — CHANGES_REQUESTED** [review]
Nice job, Siarhei! Fix some cosmetic remarks and the PR will be approved. 
Remember to ping me after fixing them.

File: StructuralPatterns/Adapter/StringAdapter.cs (hunk 3 above)
Comment (MaksimNikitin) [reply StructuralPatterns/Adapter/StringAdapter.cs]: By the way, foreach in this case would look better.

**Comment by SiarheiBarkouski** [comment]
I modified the project. Pls, have a look at it.
@MaksimNikitin

**Review by MaksimNikitin — APPROVED** [review]

**Pull request merged**